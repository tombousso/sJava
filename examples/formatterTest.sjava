(package examples.formatterTest)

(define-class Main () 'public
	((main args String[]) void 'public 'static
		(Main:printFormatted
			"""

(include
	`(
		(
			abc
		)
		a
	)( (abc\"""a\"""
	) )
`()
)(if (

	|| (||
		false
		)
		false)
	true
)"""
		)
		(Main:printFormatted
			"""
			((
				abc
				def
			):ghi 456 789)
			
		"""
		)
	)
	((printFormatted code String) void 'static
		(println (sjava.compiler.Main:formatCode code) "---")
	)
)
