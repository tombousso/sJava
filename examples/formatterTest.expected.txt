(include
	`((abc)
		a
	) ((abc """a"""))
	`()
)
(if
	(||
		(||
			false
		)
		false
	)
	true
)
---
(
	(abc
		def
	):ghi 456 789
)
---
